### PR TITLE
Adding Demo for View-Job-Filters plugin

### PR DESCRIPTION
**Pull request by dheerajodha**
This PR adds demo for the "View Job Filters" plugin.
- Added yaml file containing the configuration.
- Added README.md file
- Added Test file, to test the configuration.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: demos/view-job-filters/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            includeExcludeTypeString: "includeMatched"
+            lessThan: true
+        - buildStatusFilter:
+            building: false
+            inBuildQueue: true
+            includeExcludeTypeString: "includeMatched"
+            neverBuilt: true
+        - jobTypeFilter:
+            includeExcludeTypeString: "includeMatched"
+            jobType: "hudson.model.FreeStyleProject$DescriptorImpl@5d0507bc"
```
Comment (timja) [inline demos/view-job-filters/README.md:34]: I wouldn't say this is working, probably best to exclude it

Comment (dheerajodha) [reply demos/view-job-filters/README.md]: Sure :+1: 
I made the changes

File: demos/view-job-filters/README.md [hunk 2]
```diff
@@ -31,7 +31,6 @@ jenkins:
             neverBuilt: true
         - jobTypeFilter:
```
Comment (timja) [inline demos/view-job-filters/README.md:32]: is it working without the type though? better to just remove this filter if it's not compatible with jcasc

Comment (dheerajodha) [reply demos/view-job-filters/README.md:32]: Oh sorry, missed that :sweat_smile:  thanks for pointing it out

**Comment by dheerajodha** [comment]
I'm getting compilation errors, the classes "BuildDurationFilter", "BuildStatusFilter" and "SecurityFilter" are not found.
I tried it locally as well on Eclipse, I referred to the javadocs of hudson.views to get the path of these above-mentioned classes, so that I can import them.

For example, to get the path of class "BuildDurationFilter" I went here and copied the mentioned path on top: "hudson.views.BuildDurationFilter"

What is the correct way of importing the classes?

**Comment by timja** [comment]
> I'm getting compilation errors, the classes "BuildDurationFilter", "BuildStatusFilter" and "SecurityFilter" are not found.
> I tried it locally as well on Eclipse, I referred to the javadocs of hudson.views to get the path of these above-mentioned classes, so that I can import them.
> 
> For example, to get the path of class "BuildDurationFilter" I went [here](https://javadoc.jenkins.io/plugin/view-job-filters/hudson/views/BuildDurationFilter.html#BuildDurationFilter-float-boolean-java.lang.String-float-java.lang.String-java.lang.String-) and copied the mentioned path on top: "hudson.views.BuildDurationFilter"
> 
> What is the correct way of importing the classes?

Add a dependency on the plugin in the integrations folder: https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/integrations/pom.xml

File: integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    public void configure_view_job_filters() throws Exception {
+		
+		final BuildDurationFilter buildDurationFilter = new BuilDurationFilter();
+		assertTrue(buildDurationFilter.isLessThan());
+		assertThat(buildDurationFilter.getBuildDurationMinutes, is(5));
+		assertThat(buildDurationFilter.getAmount(), is(60.0));
+		assertThat(buildDurationFilter.getAmountTypeString(), is("Days"));
+		assertThat(buildDurationFilter.getBuildCountTypeString(), is("Latest"));
+		assertTrue(buildDurationFilter.isIncludeMatched());	
+
```
Comment (dheerajodha) [inline integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java:41]: Hi @timja, I referred to other test files to try to understand how to write tests, however, can you tell me if this is the right way to test my yaml configuration for this plugin?

Now, that I correctly imported the classes (by specifying the dependencies), I see that constructor BuildDurationFilter() is not defined. There's only one constructor and that is parameterized with all the above-mentioned values.

So I'll pass these hardcoded values into the constructor, then use the getters to fetch these values from the object and test it against what? I guess I'll test them against the values in the yaml file. If so, how do I fetch them here? using the JenkinsConfiguredWithReadmeRule object?

Comment (timja) [reply integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java:41]: you need to retrieve the view, not create a filter with the constructor

Jenkins.get().getViews() or something like that and work your way down till you find the view

**Comment by timja** [comment]
Are you still interested in this?

**Comment by dheerajodha** [comment]
Oh yes, forgot about this so I will try working on this and if I'm not able to do this within few days I'll let you know here.

File: integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    @Test
+    @ConfiguredWithReadme(value = "view-job-filters/README.md")
+    public void configure_view_job_filters() throws Exception {
+        BuildDurationFilter duration = null;
+        BuildStatusFilter status = null;
+        SecurityFilter security = null;
+
+        Jenkins jenkins = j.jenkins;
+        Collection<View> views = jenkins.getViews();
+        for (View view : views) {
```
Comment (timja) [inline integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java:37]: better to just grab `views.get(0)`, you know how many elements are in the list, and then remove the for loop

File: integrations/src/test/resources/io/jenkins/plugins/casc/ViewJobFilters.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+jenkins:
```
Comment (timja) [inline integrations/src/test/resources/io/jenkins/plugins/casc/ViewJobFilters.yml:1]: why is this file here? I don't think it's used

Comment (dheerajodha) [reply integrations/src/test/resources/io/jenkins/plugins/casc/ViewJobFilters.yml:1]: oh, I referred to other tests and they had their yaml file here along with one at their readme.md. I'll delete it now.

**Review by timja — APPROVED** [review]

File: integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java (hunk 4 above)
Comment (dheerajodha) [reply integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java]: so isn't it a problem that the [compilation failed](https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fconfiguration-as-code-plugin/detail/PR-1603/30/pipeline) with the error _cannot find symbol:   method get(int)_?

**Review by timja — APPROVED** [review]
Thanks!

**Comment by timja** [comment]
@jetersen is it possible to disable LGTM? it's getting shutdown anyway

**Comment by jetersen** [comment]
Feel free to open PR

**Comment by timja** [comment]
> Feel free to open PR

is it in this repo? I couldn't see anything

**Comment by timja** [comment]
I've removed it from https://github.com/organizations/jenkinsci/settings/installations/1577539
